src/bootstrap/src/utils/execution_context.rs

```diff
@@ -89,22 +90,23 @@ impl ExecutionContext {
     ) -> DeferredCommand<'a> {
         command.mark_as_executed();
 
+        let created_at = command.get_created_location();
+        let executed_at = std::panic::Location::caller();
```

Comment:
To clarify, `executed_at` should be created here, and the resulting `Location` should be stored in `DeferredCommand`. `crated_at` can be read in `wait_for_output` :)